### PR TITLE
fix grid, octree: faster point distribution

### PR DESCRIPTION
**Pull request by true-real-michael**
A new method of distributing points into voxels is used
1. The target voxel identifier is calculated for all points.
2. The unique identifiers are extracted.
3. Points are reordered based of the unique identifiers and split, to form groups of points, where each group is inserted into the corresponding voxel.
4. Points are inserted into voxels.

### CODE REVIEW

File: octreelib/grid/grid.py [hunk 1]
```diff
@@ -59,29 +57,35 @@ def __init__(self, grid_config: GridConfig):
...
-        unique_indices = np.unique(voxel_indices, axis=0)
-        for unique_id in unique_indices:
-            mask = np.where((voxel_indices == unique_id).all(axis=1))
-            distributed_points[tuple(unique_id)] = points[mask]
-
-        # insert points to octrees
-        for voxel_coordinates, voxel_points in distributed_points.items():
+            (points - self._grid_config.corner) // self._grid_config.voxel_edge_length
+        ) * self._grid_config.voxel_edge_length
+        voxel_indices = voxel_indices.astype(int)
```
Comment (pmokeev) [inline octreelib/grid/grid.py:74]: Наверное это можно сразу строкой выше сделать, в другой функции ты так и сделал

File: octreelib/octree/octree.py [hunk 2]
```diff
@@ -67,21 +67,27 @@ def insert_points(self, points: PointCloud):
...
+            # Reorder and group points by the child to insert in
+            grouped_points = np.split(
+                points[inverse_indices.argsort()],
+                np.cumsum(np.bincount(inverse_indices))[:-1],
+            )
+            for child_index, child_points in zip(unique_indices, grouped_points):
+                # Calculate the internal child id from its binary representation
+                child_internal_id = (
+                    child_index[0] * 4 + child_index[1] * 2 + child_index[2]
+                )
```
Comment (pmokeev) [inline octreelib/octree/octree.py:89]: Вот тут не совсем понятно что за индексы 4 и 2

**Review by pmokeev — CHANGES_REQUESTED** [review]

File: octreelib/grid/grid.py (hunk 1 above)
Comment (true-real-michael) [reply octreelib/grid/grid.py:74]: исправил

File: octreelib/octree/octree.py (hunk 2 above)
Comment (true-real-michael) [reply octreelib/octree/octree.py]: исправил

**Review by pmokeev — APPROVED** [review]

**Pull request merged**